fuel-types/Cargo.toml

```diff
@@ -15,6 +15,7 @@ fuel-derive = { workspace = true }
 hex = { version = "0.4", default-features = false, optional = true }
 rand = { version = "0.8", default-features = false, optional = true }
 serde = { version = "1.0", default-features = false, features = ["derive", "alloc"], optional = true }
+sha2 = { version = "0.10", default-features = false }
```

Comment:
One of the reasons why we didn't do that during the initial implementation is that we didn't want to increase the number of dependencies. Plus, we want to use the default hasher from `fuel-crypto`.

Reply:
I tried using fuel-crypto, but that creates a circular dependency, not sure what to do

Reply:
Even if you can, it still brings a lot of dependencies=) I meant it is why we decided to implement `asset_id` method in `fuel-tx`, because it is the business logic defined on `fuel-tx` level rather than on `fuel-types`.

Reply:
Hmm i think ill just close this pr and make a simpler one which just adds the default asset method in the ContractIdExt trait then